### PR TITLE
Update spack openfoam example to Slurm V6

### PR DESCRIPTION
**Pull request by harshthakkar01**
### Submission Checklist

Please take the following actions before submitting this pull request.

* Fork your PR branch from the Toolkit "develop" branch (not main)
* Test all changes with pre-commit in a local branch [#](https://goo.gle/hpc-toolkit#development)
* Confirm that "make tests" passes all tests
* Add or modify unit tests to cover code changes
* Ensure that unit test coverage remains above 80%
* Update all applicable documentation
* Follow Cloud HPC Toolkit Contribution guidelines [#](https://goo.gle/hpc-toolkit-contributing)

### CODE REVIEW

File: docs/tutorials/openfoam/spack-openfoam.yaml [hunk 1]
```diff
@@ -153,25 +158,37 @@ deployment_groups:
           mpirun -n 60 -npernode 30 -hostfile hostfile snappyHexMesh -overwrite -parallel
           mpirun -n 60 -npernode 30 -hostfile hostfile potentialFoam -parallel
           mpirun -n 60 -npernode 30 -hostfile hostfile simpleFoam -parallel
+
+  - id: compute_nodeset
+    source: community/modules/compute/schedmd-slurm-gcp-v6-nodeset
+    use: [network1]
+    settings:
+      name: ns
```
Comment (cboneti) [inline docs/tutorials/openfoam/spack-openfoam.yaml:166]: I believe name is no longer required, I would remove it if not required.

File: docs/tutorials/openfoam/spack-openfoam.yaml [hunk 2]
```diff
@@ -114,6 +114,11 @@ deployment_groups:
     source: modules/scripts/startup-script
     settings:
       runners:
+      - type: shell
+        destination: remove_lustre_client.sh
+        content: |
+          #!/bin/bash
+          rm /etc/yum.repos.d/lustre-client.repo
```
Comment (cboneti) [inline docs/tutorials/openfoam/spack-openfoam.yaml:121]: line 130,131, why are these lines needed now?
``` 
          mkdir -p /apps/openfoam
          chmod a+rwx /apps/openfoam
```

**Review by cboneti — CHANGES_REQUESTED** [review]
Please also fix the documentation:
1.  point to our fork of Slurm-GCP. 
2. the spack cache comment is outdated now, are we using the GCP cache by default? what is the completion time? 
3. Cluster deployment: you changed the deployment name, therefore, you have to change the deployment steps. The connection to the controller will also be affected since the cluster name, with v6, will likely be different. Same goes with the login node.

File: docs/tutorials/openfoam/spack-openfoam.yaml (hunk 2 above)
Comment (cboneti) [reply docs/tutorials/openfoam/spack-openfoam.yaml:121]: please make sure you test it.

File: docs/tutorials/openfoam/spack-openfoam.yaml [hunk 3]
```diff
@@ -14,11 +14,11 @@
 
 ---
 
-blueprint_name: spack-openfoam
+blueprint_name: spack-openfoam-v6
```
Comment (mr0re1) [inline docs/tutorials/openfoam/spack-openfoam.yaml:17]: I'm not sure if we need spell `v6` out here.

File: docs/tutorials/openfoam/spack-openfoam.yaml [hunk 4]
```diff
@@ -14,11 +14,11 @@
...
 
 vars:
   project_id:  ## Set GCP Project ID Here ##
-  deployment_name: spack-openfoam
+  deployment_name: spack-openfoam-v6
```
Comment (mr0re1) [inline docs/tutorials/openfoam/spack-openfoam.yaml:21]: same

File: docs/tutorials/openfoam/spack-openfoam.yaml (hunk 2 above)
Comment (harshthakkar01) [reply docs/tutorials/openfoam/spack-openfoam.yaml:121]: Removed and ran the test.

File: docs/tutorials/openfoam/spack-openfoam.yaml (hunk 1 above)
Comment (harshthakkar01) [reply docs/tutorials/openfoam/spack-openfoam.yaml]: Removed

File: docs/tutorials/openfoam/spack-openfoam.yaml (hunk 4 above)
Comment (harshthakkar01) [reply docs/tutorials/openfoam/spack-openfoam.yaml]: Removed.

File: docs/tutorials/openfoam/spack-openfoam.yaml (hunk 3 above)
Comment (harshthakkar01) [reply docs/tutorials/openfoam/spack-openfoam.yaml]: Removed.

File: docs/tutorials/openfoam/spack-openfoam.md [hunk 5]
```diff
@@ -106,24 +103,19 @@ contains the terraform needed to deploy your cluster.
 
 ## Deploy the Cluster
 
-Use the following commands to run terraform and deploy your cluster.
-
+Use below command to deploy your cluster.
 ```bash
-terraform -chdir=spack-openfoam/primary init
-terraform -chdir=spack-openfoam/primary apply
+./ghpc deploy spack-openfoam
```
Comment (mr0re1) [inline docs/tutorials/openfoam/spack-openfoam.md:108]: nit. Remove `./` prefix, just `ghpc`

File: docs/tutorials/openfoam/spack-openfoam.md [hunk 6]
```diff
@@ -280,7 +272,7 @@ exit
 Run the following command in the cloud shell terminal to destroy the cluster:
 
 ```bash
-terraform -chdir=spack-openfoam/primary destroy -auto-approve
+./ghpc deploy spack-openfoam
```
Comment (mr0re1) [inline docs/tutorials/openfoam/spack-openfoam.md:275]: nit. same

**Review by mr0re1 — APPROVED** [review]

**Review by jrossthomson — APPROVED** [review]

File: docs/tutorials/openfoam/spack-openfoam.yaml [hunk 7]
```diff
@@ -36,7 +36,7 @@ deployment_groups:
     source: community/modules/scripts/spack-setup
     settings:
       install_dir: /apps/spack
-      spack_ref: v0.19.0
+      spack_ref: v0.21.0
```
[the remark this answers is not in the document]
Comment (harshthakkar01) [reply docs/tutorials/openfoam/spack-openfoam.yaml]: Makes sense. Updated it to v0.20.0.

File: docs/tutorials/openfoam/spack-openfoam.md (hunk 5 above)
Comment (harshthakkar01) [reply docs/tutorials/openfoam/spack-openfoam.md:108]: After offline discussion with Ivan, updating it to local copy.

**Review by mr0re1 — APPROVED** [review]
